integers_and_arrays.rb

```diff
@@ -83,16 +111,51 @@ def intersection(array_1, array_2)
 # Assumption/ Given: All numbers in the matrix are 0s or 1s
 # If any number is found to be 0, the method updates all the numbers in the
 # corresponding row as well as the corresponding column to be 0.
-def matrix_convert_to_0(matrix)
-  puts "NOT IMPLEMENTED"
+def matrix_convert_to_0(matrix) # time complex O(n^2 + n^3) => O(n^3) (kind of, since that assumes n = m, which it doesnt), space complex O(2n) => O(n)
+  rows_with_0s = []
+  columns_with_0s = []
+  matrix.each_with_index do |row, x|
+    row.each_with_index do |item, y|
+      if item == 0
+        columns_with_0s << y
+        rows_with_0s << x
+      end
+    end
+  end
+  matrix.each_with_index do |row, x|
```

Comment:
Instead of searching in rows_with_0s and columns_with_0s, you could optimize by changing this to two parts:
`matrix.each_with_index do |row, x|`
`columns_with_0s.each do |column|`
`matrix[x][column] = 0`
`end`
`end`
`rows_with_0s.each do |row|`
`matrix[0].length.times do |column|`
`matrix[row][column] = 0`
`end`
`end`
If all rows and all columns get added to rows_with_0s and columns_with_0s then each of these two loop will get run (n*m) number of times. So in total, in the worst case it will be 3 * (n*m) or O(n*m).